### PR TITLE
[AIRFLOW-3395] Add REST API endpoints to the docs

### PR DESCRIPTION
**Pull request by xnuinside**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-3395\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3395
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

I don't see any task about start using swagger or something else, so I added temporary doc for informing about existed endpoints

I also try to use flask autodoc https://sphinxcontrib-httpdomain.readthedocs.io/en/stable/#module-sphinxcontrib.autohttp.flask but has an issue relative to defining flask app, I believe need to do some refactor to use flask app, so I just used simple httpdomain sphinx.

Added info about REST API endpoints in doc 
result doc preview: 
screen shot 2018-11-25 at 21 10 37
screen shot 2018-11-25 at 21 10 43

if try to use autofunction, for example, will be problems such as: 

''' File "/Users/jvolkova/.virtualenvs/airflow_tutorial/lib/python2.7/site-packages/airflow/www_rbac/api/experimental/endpoints.py", line 38, in <module>
    requires_authentication = airflow.api.api_auth.requires_authentication
AttributeError: 'NoneType' object has no attribute 'requires_authentication'
''' so need to refactor 'endpoints.py' to avoid those issues, but I don't want to do it right now, because as I remember we plan do something with REST API
### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by xnuinside** [comment]
@ashb, @kaxil, @feng-tao, please review

**Comment by Fokko** [comment]
Nice one @xnuinside Looks great. I've restarted the CI.

File: docs/api.rst [hunk 1]
```diff
@@ -24,10 +24,73 @@ available at /api/experimental/. Please note that we expect the endpoint definit
 Endpoints
 ---------
 
-This is a place holder until the swagger definitions are active
+.. http:post:: /api/experimental/dags/<DAG_ID>/dag_runs
+
+  Creates a dag_run for a given dag id (POST).
```
Comment (Fokko) [inline docs/api.rst]: Isn't the (POST) a bit redundant?

File: docs/api.rst [hunk 2]
```diff
@@ -24,10 +24,73 @@ available at /api/experimental/. Please note that we expect the endpoint definit
...
+
+
+.. http:get:: /api/experimental/test
+
+  To check REST API server correct work. Return status 'OK'.
+
+
+.. http:get:: /api/experimental/dags/<DAG_ID>/tasks/<TASK_ID>
+
+  Returns info for a task (GET).
```
Comment (Fokko) [inline docs/api.rst]: Same here.

**Review by Fokko — COMMENTED** [review]
Looks good. We could also consider adding Swagger support to the experimental endpoint: https://github.com/soerface/flask-restful-swagger-2.0

**Comment by xnuinside** [comment]
@Fokko, fixed . about https://github.com/soerface/flask-restful-swagger-2.0 I think we could talk about it as an adding new feature with a separate task (something told to me, what it is not quick to add and, at first, need to decide what we want use swagger)

**Comment by xnuinside** [comment]
@Fokko, two tests failed with strange errors, I believe it's not relative to PR )

**Comment by xnuinside** [comment]
@Fokko, tests fail with 'Attempting to fetch rat
rm: cannot remove '/tmp/lib/apache-rat-0.12.jar': No such file or directory
Our attempt to download rat locally to /tmp/lib/apache-rat-0.12.jar failed. Please install rat manually.
ERROR: InvocationError for command '/app/scripts/ci/6-check-license.sh' (exited with code 255)'

**Comment by Fokko** [comment]
@xnuinside Some weird stuff happening with the CI, I've restarted them. Let's see what happens.

**Comment by ron819** [comment]
@Fokko  The lib https://github.com/soerface/flask-restful-swagger-2.0 isn't maintained.

Better use https://github.com/noirbizarre/flask-restplus  ?

**Comment by ron819** [comment]
@Fokko  The lib https://github.com/soerface/flask-restful-swagger-2.0 isn't maintained.

   Better use https://github.com/noirbizarre/flask-restplus  ?

**Comment by xnuinside** [comment]
@Fokko, yeeee ) successful Travis build

**Comment by Fokko** [comment]
> @Fokko The lib https://github.com/soerface/flask-restful-swagger-2.0 isn't maintained.
> 
> Better use https://github.com/noirbizarre/flask-restplus ?

Good point, I'm not really into the different swagger extensions for Flask, so the `flask-restful-swagger-2.0` was just a suggestion. `flask-restplus` sounds like a better alternative.

**Comment by Fokko** [comment]
Thanks @xnuinside :-)

**Comment by ashb** [comment]
Swagger is now rebranded and re-badged as OpenAPI (Sometimes also called Swagger v3) so it would be nice to find a module supporting that:

https://smartbear.com/blog/develop/what-is-the-difference-between-swagger-and-openapi/

**Pull request merged**